lib/middleware/csrfProtected.ts

```diff
@@ -3,21 +3,22 @@ import { MiddlewareRequest, MiddlewareReturn } from "@lib/types";
 import { getCsrfToken } from "next-auth/client";
 import { logMessage } from "@lib/logger";
 
-export const csrfProtected = (allowMethods: string[]): MiddlewareRequest => {
+export const csrfProtected = (protectedMethods: string[]): MiddlewareRequest => {
   return async (req: NextApiRequest, res: NextApiResponse): Promise<MiddlewareReturn> => {
     try {
-      if (allowMethods && req.method && allowMethods.includes(req.method)) {
+      if (isProtected(req, protectedMethods)) {
```

Comment:
This logic can be simplified a bit.  If a CSRF header is not present it should be treated as unauthorized and not throw a 500 server error.  If as Cross site attack vector were to happen and a CSRF token was not provided the current logic would flood our logs with Errors and trigger Application Alarms that something was wrong with the site.  In this case the site is behaving normally by blocking non csrf and invalid csrf headers.
```js
    try {
      if (isProtected(req, protectedMethods)) {
        const csrfToken = await getCsrfToken({ req });

        if (csrfToken && csrfToken === req.headers["x-csrf-token"]) {
          return { next: true };
        } else {
          res.status(403).json({ error: "Access Denied" });
          return { next: false };
        }
      } else {
        //allow unrestricted method
        return { next: true };
      }
    } catch (error) {
      logMessage.error(error);
      res.status(500).json({ error: "Malformed API Request" });
      return { next: false };
    }
```